data/utils/defectsFromBias.py

```diff
@@ -112,13 +46,20 @@ def writeDefectsFile(bboxList, path, detectorSerial, detectorName):
     args = parser.parse_args()
 
     biasMI = afwImage.MaskedImageF(args.bias)
-    defectList = getDefectListFromMask(biasMI, "BAD")
-    bboxList = [defect.getBBox() for defect in defectList]
-    writeDefectsFile(bboxList, DefectsPath, detectorSerial, detectorName)
+    defectList = Defects.fromMask(biasMI, "BAD")
+    valid_start = date_parser.parse('19700101T000000')
+    md = defectList.getMetadata()
+    md['INSTRUME'] = 'test'
+    md['DETECTOR'] = detectorName
+    md['CALIBDATE'] = valid_start.isoformat()
+    md['FILTER'] = None
+    md['CALIB_ID'] = (f'detector={detectorName} calibDate={valid_start.isoformat()} '
+                      'ccd={detectorName} ccdnum={detectorName} filter=None')
+    defectList.writeText(DefectsPath)
     print("wrote defects file %r" % (DefectsPath,))
 
-    test2BBoxList = getBBoxList(DefectsPath, detectorName)
-    assert len(bboxList) == len(test2BBoxList)
-    for boxA, boxB in zip(bboxList, test2BBoxList):
-        assert boxA == boxB
+    test2defectList = Defects.readText(DefectsPath+".ecsv")
+    assert len(defectList) == len(test2defectList)
```

Comment:
I think you should use simple equality of defect list here.  `Defects` defines an `__eq__` method for exactly this purpose so you don't have to rely on the internals of how `Defect` is implemented.

Reply:
Agreed, if that works. The two sequences would have to be of the same type (which you can enforce using tuple(sequence) or list(sequence) if necessary).

Reply:
They aren't sequences as such. They are meas_algorithms `Defects` which have a defined `__eq__`.

The `List` in the name is possibly confusing but `Defects` act like a `list`.

Reply:
Done.